framework/source/class/qx/ui/window/Window.js

```diff
@@ -1050,5 +1210,29 @@ qx.Class.define("qx.ui.window.Window",
       this.close();
       this.getChildControl("close-button").reset();
     }
+  },
+
+  destruct : function()
+  {
+    var id;
+    var parent;
+
+    // Remove ourselves from the focus handler
+    qx.ui.core.FocusHandler.getInstance().removeRoot(this);
+
+    // Remove the listener for appear, if there is one
+    id = this.__centeringAppearId;
+    id && this.removeListenerById(id);
+
+    // If we haven't been removed from our parent, clean it up too.
+    parent = this.getLayoutParent();
+    if (parent) {
```

Comment:
If the logic for attaching/detaching the resize listener to the parent is moved to ```setLayoutParent``` we could simply call here ```this.setLayoutParent(null);``` which removes the parent listener if it's there.